dotCMS/src/main/java/com/dotcms/vanityurl/handler/DefaultVanityUrlHandler.java

```diff
@@ -0,0 +1,95 @@
+package com.dotcms.vanityurl.handler;
+
+import com.dotcms.vanityurl.model.CachedVanityUrl;
+import com.dotcms.vanityurl.model.VanityUrlResult;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.filters.CMSFilter;
+import com.dotmarketing.filters.CmsUrlUtil;
+import com.dotmarketing.util.InodeUtils;
+import com.dotmarketing.util.UtilMethods;
+import java.io.IOException;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This class implements the methods defined in the {@link VanityUrlHandler}
+ *
+ * @author oswaldogallango
+ * @version 4.2.0
+ * @since June 16, 2017
+ */
+public class DefaultVanityUrlHandler implements VanityUrlHandler {
+
+    private static final VanityUrlResult DEFAULT_RESULT = new VanityUrlResult(null, null,
+            CMSFilter.IAm.NOTHING_IN_THE_CMS, true);
+
+    private static final CmsUrlUtil urlUtil = CmsUrlUtil.getInstance();
+
+    @Override
+    public VanityUrlResult handle(final CachedVanityUrl vanityUrl,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)